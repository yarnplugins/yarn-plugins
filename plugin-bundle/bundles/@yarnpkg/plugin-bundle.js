/* eslint-disable */
module.exports = {
name: "@yarnpkg/plugin-bundle",
factory: function (require) {
var plugin;plugin=(()=>{"use strict";var e={543:(e,t,r)=>{r.r(t),r.d(t,{default:()=>p});var n=r(42),o=r(966),i=r(594),c=r(688),a=r(76),s=function(e,t,r,n){var o,i=arguments.length,c=i<3?t:null===n?n=Object.getOwnPropertyDescriptor(t,r):n;if("object"==typeof Reflect&&"function"==typeof Reflect.decorate)c=Reflect.decorate(e,t,r,n);else for(var a=e.length-1;a>=0;a--)(o=e[a])&&(c=(i<3?o(c):i>3?o(t,r,c):o(t,r))||c);return i>3&&c&&Object.defineProperty(t,r,c),c};class l extends n.Command{async execute(){const e=await o.Configuration.find(this.context.cwd,this.context.plugins),{project:t,workspace:r}=await o.Project.find(e,this.context.cwd);if(!r)throw new i.WorkspaceRequiredError(t.cwd,this.context.cwd);const n=c.ppath.resolve(r.cwd,"package.zip");return(await o.StreamReport.start({configuration:e,stdout:this.context.stdout},async o=>{var i;o.reportJson({base:r.cwd});const s=await(0,a.getLibzipPromise)(),l=new c.ZipFS(n,{create:!0,libzip:s}),p=new c.JailFS(r.cwd),d=e.get("yarnPath"),u=c.ppath.relative(t.cwd,d);o.reportInfo(null,u),o.reportJson({location:u}),l.mkdirSync(".yarn"),l.mkdirSync(".yarn/releases");const f=c.xfs.readFileSync(d);l.writeFileSync(u,f);const y=["lib","package.json"];for(;y.length>0;){const e=y.shift();if(p.lstatSync(e).isDirectory())l.mkdirSync(e),p.readdirSync(e).forEach(t=>{y.unshift(c.ppath.join(e,t))});else{o.reportInfo(null,e),o.reportJson({location:e});const t=c.ppath.resolve(r.cwd,e);if("package.json"===e){const r=JSON.parse(c.xfs.readFileSync(t,"utf8"));delete r.devDependencies,delete r.peerDependencies,delete r.bundledDependencies,delete r.jest,Object.keys(r).filter(e=>e.endsWith("Config")).forEach(e=>{delete r[e]}),r.main&&(null===(i=r.scripts)||void 0===i?void 0:i.start)&&(r.scripts.start="node "+r.main),Object.keys(r.scripts||{}).forEach(e=>{"start"!==e&&"postinstall"!==e&&delete r.scripts[e]});const n=JSON.stringify(r,null,"  ");l.writeFileSync(e,n,"utf8")}else{const r=c.xfs.readFileSync(t);l.writeFileSync(e,r)}}}o.reportInfo(null,".yarnrc.yml"),o.reportJson({location:".yarnrc.yml"}),l.writeFileSync(".yarnrc.yml",`yarnPath: ${u}\n`,"utf8"),l.saveAndClose()})).exitCode()}}s([n.Command.Path("bundle")],l.prototype,"execute",null);const p={commands:[l]}},594:e=>{e.exports=require("@yarnpkg/cli")},966:e=>{e.exports=require("@yarnpkg/core")},688:e=>{e.exports=require("@yarnpkg/fslib")},76:e=>{e.exports=require("@yarnpkg/libzip")},42:e=>{e.exports=require("clipanion")}},t={};function r(n){if(t[n])return t[n].exports;var o=t[n]={exports:{}};return e[n](o,o.exports,r),o.exports}return r.n=e=>{var t=e&&e.__esModule?()=>e.default:()=>e;return r.d(t,{a:t}),t},r.d=(e,t)=>{for(var n in t)r.o(t,n)&&!r.o(e,n)&&Object.defineProperty(e,n,{enumerable:!0,get:t[n]})},r.o=(e,t)=>Object.prototype.hasOwnProperty.call(e,t),r.r=e=>{"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(e,"__esModule",{value:!0})},r(543)})();
return plugin;
}
};