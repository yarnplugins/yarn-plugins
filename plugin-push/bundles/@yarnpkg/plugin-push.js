/* eslint-disable */
module.exports = {
name: "@yarnpkg/plugin-push",
factory: function (require) {
var plugin;plugin=(()=>{var t={543:(t,e,r)=>{"use strict";r.r(e),r.d(e,{PushCommand:()=>l,default:()=>f});var n=r(87),o=r.n(n),s=r(725),i=r.n(s),c=r(966),a=r(688),p=r(42),u=function(t,e,r,n){var o,s=arguments.length,i=s<3?e:null===n?n=Object.getOwnPropertyDescriptor(e,r):n;if("object"==typeof Reflect&&"function"==typeof Reflect.decorate)i=Reflect.decorate(t,e,r,n);else for(var c=t.length-1;c>=0;c--)(o=t[c])&&(i=(s<3?o(i):s>3?o(e,r,i):o(e,r))||i);return s>3&&i&&Object.defineProperty(e,r,i),i};class l extends p.Command{constructor(){super(...arguments),this.bucket=process.env.PKG_BUCKET||"",this.version=o().userInfo().username,this.files=[]}async execute(){const{cwd:t}=this.context,e=await c.Configuration.find(t,null),{workspace:r}=await c.Project.find(e,t);if(!this.bucket)throw new p.UsageError("Need to pass a cloud storage bucket, e.g. --bucket=gs://pkg.example.com");const n=this.bucket.includes("://")?this.bucket.endsWith("/")?this.bucket:this.bucket+"/":this.bucket.endsWith("/")?"gs://"+this.bucket:`gs://${this.bucket}/`;if(!n.startsWith("gs://"))throw new Error("Only Google Cloud Storage buckets are supported at this moment.");if(0===this.files.length)throw new p.UsageError("Need to pass the list of files.");const s=[];for(const t of this.files){const e=a.ppath.resolve(t);if(!await a.xfs.existsPromise(e))throw new Error("File not found: "+e);if(!r||!r.manifest.name)throw new Error("Failed to read package.json");const o=a.ppath.extname(t),i=a.ppath.basename(t).slice(0,-o.length),c=1===this.files.length?`${r.manifest.name.name}_${this.version}${o}`:`${r.manifest.name.name}_${i}_${this.version}${o}`;s.push({name:t,source:e,target:`${n}${c}`})}for(const{name:t,source:e,target:r}of s){this.context.stdout.write(`${t} -> ${r}${o().EOL}`);const n=await new Promise((t,n)=>{i()("gsutil",["cp",e,r],{cwd:this.context.cwd,stdio:[this.context.stdin,this.context.stdout,this.context.stderr]}).on("error",t=>{n(t)}).on("close",e=>{t(e)})});if(0!==n)return n}}}l.usage=p.Command.Usage({category:"deployment",description:"Uploads application bundle to a cloud storage bucket.",examples:[["Upload package.zip to gs://pkg.example.com/{pkgName}_123.zip","$0 push --bucket=pkg.example.com --version=123 package.zip"],["Upload multiple files","$0 push --bucket=pkg.example.com --version=123 one.js two.js"]]}),u([p.Command.String("--bucket")],l.prototype,"bucket",void 0),u([p.Command.String("--version")],l.prototype,"version",void 0),u([p.Command.Rest()],l.prototype,"files",void 0),u([p.Command.Path("push")],l.prototype,"execute",null);const f={commands:[l]}},725:(t,e,r)=>{"use strict";const n=r(129),o=r(626),s=r(70);function i(t,e,r){const i=o(t,e,r),c=n.spawn(i.command,i.args,i.options);return s.hookChildProcess(c,i),c}t.exports=i,t.exports.spawn=i,t.exports.sync=function(t,e,r){const i=o(t,e,r),c=n.spawnSync(i.command,i.args,i.options);return c.error=c.error||s.verifyENOENTSync(c.status,i),c},t.exports._parse=o,t.exports._enoent=s},70:t=>{"use strict";const e="win32"===process.platform;function r(t,e){return Object.assign(new Error(`${e} ${t.command} ENOENT`),{code:"ENOENT",errno:"ENOENT",syscall:`${e} ${t.command}`,path:t.command,spawnargs:t.args})}function n(t,n){return e&&1===t&&!n.file?r(n.original,"spawn"):null}t.exports={hookChildProcess:function(t,r){if(!e)return;const o=t.emit;t.emit=function(e,s){if("exit"===e){const e=n(s,r);if(e)return o.call(t,"error",e)}return o.apply(t,arguments)}},verifyENOENT:n,verifyENOENTSync:function(t,n){return e&&1===t&&!n.file?r(n.original,"spawnSync"):null},notFoundError:r}},626:(t,e,r)=>{"use strict";const n=r(622),o=r(598),s=r(103),i=r(808),c="win32"===process.platform,a=/\.(?:com|exe)$/i,p=/node_modules[\\/].bin[\\/][^\\/]+\.cmd$/i;function u(t){if(!c)return t;const e=function(t){t.file=o(t);const e=t.file&&i(t.file);return e?(t.args.unshift(t.file),t.command=e,o(t)):t.file}(t),r=!a.test(e);if(t.options.forceShell||r){const r=p.test(e);t.command=n.normalize(t.command),t.command=s.command(t.command),t.args=t.args.map(t=>s.argument(t,r));const o=[t.command].concat(t.args).join(" ");t.args=["/d","/s","/c",`"${o}"`],t.command=process.env.comspec||"cmd.exe",t.options.windowsVerbatimArguments=!0}return t}t.exports=function(t,e,r){e&&!Array.isArray(e)&&(r=e,e=null);const n={command:t,args:e=e?e.slice(0):[],options:r=Object.assign({},r),file:void 0,original:{command:t,args:e}};return r.shell?n:u(n)}},103:t=>{"use strict";const e=/([()\][%!^"`<>&|;, *?])/g;t.exports.command=function(t){return t=t.replace(e,"^$1")},t.exports.argument=function(t,r){return t=(t=`"${t=(t=(t=""+t).replace(/(\\*)"/g,'$1$1\\"')).replace(/(\\*)$/,"$1$1")}"`).replace(e,"^$1"),r&&(t=t.replace(e,"^$1")),t}},808:(t,e,r)=>{"use strict";const n=r(747),o=r(383);t.exports=function(t){const e=Buffer.alloc(150);let r;try{r=n.openSync(t,"r"),n.readSync(r,e,0,150,0),n.closeSync(r)}catch(t){}return o(e.toString())}},598:(t,e,r)=>{"use strict";const n=r(622),o=r(765),s=r(463);function i(t,e){const r=t.options.env||process.env,i=process.cwd(),c=null!=t.options.cwd,a=c&&void 0!==process.chdir&&!process.chdir.disabled;if(a)try{process.chdir(t.options.cwd)}catch(t){}let p;try{p=o.sync(t.command,{path:r[s({env:r})],pathExt:e?n.delimiter:void 0})}catch(t){}finally{a&&process.chdir(i)}return p&&(p=n.resolve(c?t.options.cwd:"",p)),p}t.exports=function(t){return i(t)||i(t,!0)}},749:(t,e,r)=>{var n;r(747);function o(t,e,r){if("function"==typeof e&&(r=e,e={}),!r){if("function"!=typeof Promise)throw new TypeError("callback not provided");return new Promise((function(r,n){o(t,e||{},(function(t,e){t?n(t):r(e)}))}))}n(t,e||{},(function(t,n){t&&("EACCES"===t.code||e&&e.ignoreErrors)&&(t=null,n=!1),r(t,n)}))}n="win32"===process.platform||global.TESTING_WINDOWS?r(342):r(92),t.exports=o,o.sync=function(t,e){try{return n.sync(t,e||{})}catch(t){if(e&&e.ignoreErrors||"EACCES"===t.code)return!1;throw t}}},92:(t,e,r)=>{t.exports=o,o.sync=function(t,e){return s(n.statSync(t),e)};var n=r(747);function o(t,e,r){n.stat(t,(function(t,n){r(t,!t&&s(n,e))}))}function s(t,e){return t.isFile()&&function(t,e){var r=t.mode,n=t.uid,o=t.gid,s=void 0!==e.uid?e.uid:process.getuid&&process.getuid(),i=void 0!==e.gid?e.gid:process.getgid&&process.getgid(),c=parseInt("100",8),a=parseInt("010",8),p=parseInt("001",8),u=c|a;return r&p||r&a&&o===i||r&c&&n===s||r&u&&0===s}(t,e)}},342:(t,e,r)=>{t.exports=s,s.sync=function(t,e){return o(n.statSync(t),t,e)};var n=r(747);function o(t,e,r){return!(!t.isSymbolicLink()&&!t.isFile())&&function(t,e){var r=void 0!==e.pathExt?e.pathExt:process.env.PATHEXT;if(!r)return!0;if(-1!==(r=r.split(";")).indexOf(""))return!0;for(var n=0;n<r.length;n++){var o=r[n].toLowerCase();if(o&&t.substr(-o.length).toLowerCase()===o)return!0}return!1}(e,r)}function s(t,e,r){n.stat(t,(function(n,s){r(n,!n&&o(s,t,e))}))}},463:t=>{"use strict";const e=(t={})=>{const e=t.env||process.env;return"win32"!==(t.platform||process.platform)?"PATH":Object.keys(e).reverse().find(t=>"PATH"===t.toUpperCase())||"Path"};t.exports=e,t.exports.default=e},383:(t,e,r)=>{"use strict";const n=r(557);t.exports=(t="")=>{const e=t.match(n);if(!e)return null;const[r,o]=e[0].replace(/#! ?/,"").split(" "),s=r.split("/").pop();return"env"===s?o:o?`${s} ${o}`:s}},557:t=>{"use strict";t.exports=/^#!(.*)/},765:(t,e,r)=>{const n="win32"===process.platform||"cygwin"===process.env.OSTYPE||"msys"===process.env.OSTYPE,o=r(622),s=n?";":":",i=r(749),c=t=>Object.assign(new Error("not found: "+t),{code:"ENOENT"}),a=(t,e)=>{const r=e.colon||s,o=t.match(/\//)||n&&t.match(/\\/)?[""]:[...n?[process.cwd()]:[],...(e.path||process.env.PATH||"").split(r)],i=n?e.pathExt||process.env.PATHEXT||".EXE;.CMD;.BAT;.COM":"",c=n?i.split(r):[""];return n&&-1!==t.indexOf(".")&&""!==c[0]&&c.unshift(""),{pathEnv:o,pathExt:c,pathExtExe:i}},p=(t,e,r)=>{"function"==typeof e&&(r=e,e={}),e||(e={});const{pathEnv:n,pathExt:s,pathExtExe:p}=a(t,e),u=[],l=r=>new Promise((s,i)=>{if(r===n.length)return e.all&&u.length?s(u):i(c(t));const a=n[r],p=/^".*"$/.test(a)?a.slice(1,-1):a,l=o.join(p,t),d=!p&&/^\.[\\\/]/.test(t)?t.slice(0,2)+l:l;s(f(d,r,0))}),f=(t,r,n)=>new Promise((o,c)=>{if(n===s.length)return o(l(r+1));const a=s[n];i(t+a,{pathExt:p},(s,i)=>{if(!s&&i){if(!e.all)return o(t+a);u.push(t+a)}return o(f(t,r,n+1))})});return r?l(0).then(t=>r(null,t),r):l(0)};t.exports=p,p.sync=(t,e)=>{e=e||{};const{pathEnv:r,pathExt:n,pathExtExe:s}=a(t,e),p=[];for(let c=0;c<r.length;c++){const a=r[c],u=/^".*"$/.test(a)?a.slice(1,-1):a,l=o.join(u,t),f=!u&&/^\.[\\\/]/.test(t)?t.slice(0,2)+l:l;for(let t=0;t<n.length;t++){const r=f+n[t];try{if(i.sync(r,{pathExt:s})){if(!e.all)return r;p.push(r)}}catch(t){}}}if(e.all&&p.length)return p;if(e.nothrow)return null;throw c(t)}},966:t=>{"use strict";t.exports=require("@yarnpkg/core")},688:t=>{"use strict";t.exports=require("@yarnpkg/fslib")},129:t=>{"use strict";t.exports=require("child_process")},42:t=>{"use strict";t.exports=require("clipanion")},747:t=>{"use strict";t.exports=require("fs")},87:t=>{"use strict";t.exports=require("os")},622:t=>{"use strict";t.exports=require("path")}},e={};function r(n){if(e[n])return e[n].exports;var o=e[n]={exports:{}};return t[n](o,o.exports,r),o.exports}return r.n=t=>{var e=t&&t.__esModule?()=>t.default:()=>t;return r.d(e,{a:e}),e},r.d=(t,e)=>{for(var n in e)r.o(e,n)&&!r.o(t,n)&&Object.defineProperty(t,n,{enumerable:!0,get:e[n]})},r.o=(t,e)=>Object.prototype.hasOwnProperty.call(t,e),r.r=t=>{"undefined"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(t,Symbol.toStringTag,{value:"Module"}),Object.defineProperty(t,"__esModule",{value:!0})},r(543)})();
return plugin;
}
};