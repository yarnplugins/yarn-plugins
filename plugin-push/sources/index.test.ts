import getStream from "get-stream";
import { CommandContext, PluginConfiguration } from "@yarnpkg/core";
import { PortablePath, ppath } from "@yarnpkg/fslib";
import { PassThrough } from "stream";
import { Cli } from "clipanion";
import { PushCommand } from ".";

async function run(args: string[] = []) {
  const stream = new PassThrough();
  const cli = Cli.from<CommandContext>([PushCommand]);

  const exitCode = await cli.run(args, {
    cwd: ppath.resolve(__dirname as PortablePath),
    plugins: {} as PluginConfiguration,
    quiet: false,
    stdin: process.stdin,
    stdout: stream,
    stderr: stream,
  });

  stream.end();

  const output = await getStream(stream);

  if (exitCode !== 0) {
    throw new Error(output);
  }

  return output;
}

// TODO: Add more tests

it("Throws an error when no bucket was provided", async () => {
  expect.assertions(1);
  try {
    await run([]);
  } catch (error) {
    expect(error.message).toMatchSnapshot();
  }
});
