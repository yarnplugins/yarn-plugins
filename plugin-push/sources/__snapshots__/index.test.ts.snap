// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Throws an error when no bucket was provided 1`] = `
"Usage Error: Need to pass a cloud storage bucket, e.g. --bucket=gs://pkg.example.com

$ ...
"
`;
